src/Stratis.Bitcoin/Consensus/ChainedHeaderTree.cs

```diff
@@ -143,7 +143,7 @@ public void DisconnectPeer(int networkPeerId)
             if (!this.peerTipsByPeerId.TryGetValue(networkPeerId, out uint256 peerTipHash)) 
             {
                 this.logger.LogTrace("(-)[PEER_TIP_NOT_FOUND]");
-                throw new ConnectHeaderException();
+                return;
```

Comment:
Are we sure? last dicussion with @Aprogiena we concluded a peer must exist if Disconnect is called but we where not sure.

Reply:
This was done in case peer connects and disconnects right away. 
Another reason is in case partial or full validation fails and we remove the whole subtree so peerTipsByPeerId won't have a tip that is on an invalid chain. Old code would throw there.